File: app/src/main/java/vn/loitp/app/activity/customviews/textview/countdown/CountDownActivity.java
package vn.loitp.app.activity.customviews.textview.countdown;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import vn.loitp.app.activity.customviews.textview.countdown.lib.LCountDownView;
import vn.loitp.app.base.BaseActivity;
import vn.loitp.livestar.R;

public class CountDownActivity extends BaseActivity {

    private Button btStart;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        btStart = (Button) findViewById(R.id.bt_start);

        LCountDownView lCountDownView = (LCountDownView) findViewById(R.id.l_countdown);
        lCountDownView.setShowOrHide(false);
        lCountDownView.setCallback(new LCountDownView.Callback() {
            @Override
            public void onTick() {
                //do sth here
            }

            @Override
            public void onEnd() {
                btStart.setEnabled(true);
                lCountDownView.setShowOrHide(false);
            }
        });

        btStart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                btStart.setEnabled(false);
                lCountDownView.setShowOrHide(true);
                lCountDownView.start(5);
            }
        });
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected Activity setActivity() {
        return this;
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_count_down_textview;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/textview/countDown/CountDownActivity.kt
package vn.loitp.app.activity.customviews.textview.countDown

import android.os.Bundle
import com.loitp.annotation.IsFullScreen
import com.loitp.annotation.LogTag
import com.loitp.core.base.BaseFontActivity
import com.loitp.core.ext.setSafeOnClickListener
import com.loitp.core.utilities.LUIUtil
import com.loitpcore.views.textView.countDown.LCountDownView
import kotlinx.android.synthetic.main.activity_text_view_count_down.*
import vn.loitp.app.R

@LogTag("CountDownActivity")
@IsFullScreen(false)
class CountDownActivity : BaseFontActivity() {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_text_view_count_down
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBaseBackPressed()
                }
            )
            this.ivIconRight?.setImageResource(R.color.transparent)
            this.tvTitle?.text = CountDownActivity::class.java.simpleName
        }
        countDownView.setShowOrHide(false)
        countDownView.setCallback(object : LCountDownView.Callback {
            override fun onTick() {
                // do sth here
            }

            override fun onEnd() {
                btStart.isEnabled = true
                countDownView.setShowOrHide(false)
            }
        })

        btStart.setSafeOnClickListener {
            btStart.isEnabled = false
            countDownView.setShowOrHide(true)
            countDownView.start(5)
        }
    }
}
